lib/conversion.c

```diff
@@ -0,0 +1,204 @@
+    else if (format == pubkey_format_tss) {
+        // save raw as is
+        return files_save_bytes_to_file(path, (UINT8 *)public, sizeof(*public));
+    }
+
+    LOG_ERR("Unsupported public key output format.");
+    return false;
+}
+
+static bool tpm2_convert_pubkey_ssl(TPMT_PUBLIC *public, pubkey_format format, const char *path) {
+    bool ret = false;
+    FILE *fp = NULL;
+    RSA *ssl_rsa_key = NULL;
+
+    // need this before the first SSL call for getting human readable error
+    // strings in print_ssl_error()
+    ERR_load_crypto_strings();
+
+    if (public->type != TPM_ALG_RSA) {
+        LOG_ERR("Unsupported key type for requested output format. Only RSA is supported.");
+        goto error;
+    }
+
+    UINT32 exponent = (public->parameters).rsaDetail.exponent;
+
+    if (exponent == 0) {
+        exponent = RSA_DEFAULT_PUBLIC_EXPONENT;
+    }
+
+    // OpenSSL expects this in network byte order
+    exponent = tpm2_util_hton_32(exponent);
+
+    ssl_rsa_key = RSA_new();
+
+    if (!ssl_rsa_key) {
+        print_ssl_error("Failed to allocate OpenSSL RSA structure");
+        goto error;
+    }
+
+    ssl_rsa_key->e = BN_bin2bn((void*)&exponent, sizeof(exponent), NULL);
+    ssl_rsa_key->n = BN_bin2bn(
+            (public->unique).rsa.t.buffer,
+            (public->unique).rsa.t.size,
+            NULL);
+
+    if (!ssl_rsa_key->n || !ssl_rsa_key->e) {
+        print_ssl_error("Failed to convert data to SSL internal format");
+        goto error;
+    }
+
+    fp = fopen(path, "wb");
+
+    if (!fp) {
+        LOG_ERR("Failed to open public key output file '%s': %s",
+            path, strerror(errno));
+        goto error;
+    }
+
+    int ssl_res = 0;
+
+    switch(format) {
+    case pubkey_format_pem:
+        ssl_res = PEM_write_RSA_PUBKEY(fp, ssl_rsa_key);
+        break;
+    case pubkey_format_der:
+        ssl_res = i2d_RSA_PUBKEY_fp(fp, ssl_rsa_key);
+        break;
+    default:
+        break;
+    }
+
+    if (ssl_res <= 0) {
+        print_ssl_error("OpenSSL public key conversion failed");
+        goto error;
+    }
+
+    ret = true;
+
+error:
+    if (fp) {
+        fclose(fp);
+    }
+    if (ssl_rsa_key) {
+        RSA_free(ssl_rsa_key);
+    }
+    ERR_free_strings();
+
+    return ret;
+}
+
+bool tpm2_convert_signature(TPMT_SIGNATURE *signature, signature_format format, const char *path) {
+
+    if (format == signature_format_tss) {
+        /* TODO fix serialization */
+        return files_save_bytes_to_file(path, (UINT8 *)signature,
+            sizeof(*signature));
+    }
+    else if (format == signature_format_plain) {
+        UINT8 *buffer;
+        UINT16 size;
+
+        buffer = tpm2_extract_plain_signature(&size, signature);
+        if (buffer == NULL) {
+            return false;
+        }
+
+        bool ret = files_save_bytes_to_file(path, buffer, size);
+        free(buffer);
+
+        return ret;
+    }
+
+    LOG_ERR("Unsupported signature output format.");
+    return false;
```

Comment:
Yeah this is a hard error here, but not on line https://github.com/01org/tpm2-tools/pull/485/files#diff-98bdc2e9547136dd6f16812a7289b524R157